sdk/azcore/arm/internal/resource/resource_identifier.go

```diff
@@ -225,3 +225,15 @@ func splitStringAndOmitEmpty(v, sep string) []string {
 
 	return r
 }
+
+func isTenantResource(resourceID *ResourceID) bool {
+	return resourceID != nil && strings.EqualFold(resourceID.ResourceType.String(), TenantResourceType.String()) && resourceID.Parent == nil
```

Comment:
Can we get some comments in all these helpers that explain how they come to their "conclusion"?

Reply:
Added.